sm_timing/include/sm/timing/implementation/TimestampCorrector.hpp

```diff
@@ -67,70 +61,72 @@ typename TimestampCorrector<T>::time_t TimestampCorrector<T>::correctTimestamp(c
+               "The timestamp correction requires at least two data points "
+               "before this funciton can be called");
 
   // Get the line at the time midpoint.
-  const Point & l1 = _convexHull[_midpointSegmentIndex];
-  const Point & l2 = _convexHull[_midpointSegmentIndex + 1];
+  const Point& l1 = _convexHull[_midpointSegmentIndex];
+  const Point& l2 = _convexHull[_midpointSegmentIndex + 1u];
 
-  // look up the local timestamp
-  return  double(l2.y - l1.y) / double(l2.x - l1.x); 
+  // Look up the local timestamp.
+  return  double(l2.y - l1.y) / double(l2.x - l1.x);
 }
 
 template<typename T>
 double TimestampCorrector<T>::getOffset() const {
-  SM_ASSERT_GE(NotInitializedException, _convexHull.size(), 2, "The timestamp correction requires at least two data points before this funciton can be called");
+  SM_ASSERT_GE(NotInitializedException, _convexHull.size(), 2u,
+               "The timestamp correction requires at least two data points "
+               "before this function can be called");
   // Get the line at the time midpoint.
-  const Point & l1 = _convexHull[_midpointSegmentIndex];
-  const Point & l2 = _convexHull[_midpointSegmentIndex + 1];
+  const Point& l1 = _convexHull[_midpointSegmentIndex];
+  const Point& l2 = _convexHull[_midpointSegmentIndex + 1u];
   
-  // look up the local timestamp
-  return  double(l1.y) + (double(-l1.x) * double(l2.y - l1.y) / double(l2.x - l1.x) ); 
+  // Look up the local timestamp.
+  return  double(l1.y) + (double(-l1.x) * double(l2.y - l1.y) / double(l2.x - l1.x) );
 }
 
   
 // Get the local time from the remote time.
 template<typename T>
-typename TimestampCorrector<T>::time_t TimestampCorrector<T>::getLocalTime(const time_t & remoteTime) const
-{
-  SM_ASSERT_GE(NotInitializedException, _convexHull.size(), 2, "The timestamp correction requires at least two data points before this funciton can be called");
+typename TimestampCorrector<T>::time_t TimestampCorrector<T>::getLocalTime(
+    const time_t& remoteTime) const {
+  SM_ASSERT_GE(NotInitializedException, _convexHull.size(), 2u,
+               "The timestamp correction requires at least two data "
+               "points before this funciton can be called");
 
   // Get the line at the time midpoint.
-  const Point & l1 = _convexHull[_midpointSegmentIndex];
-  const Point & l2 = _convexHull[_midpointSegmentIndex + 1];
+  const Point& l1 = _convexHull[_midpointSegmentIndex];
+  const Point& l2 = _convexHull[_midpointSegmentIndex + 1];
 
-  // look up the local timestamp
-  return  l1.y + ((remoteTime - l1.x) * (l2.y - l1.y) / (l2.x - l1.x) ); 
-}
+  // Look up the local timestamp.
+  SM_ASSERT_GT(NotInitializedException, l2.x - l1.x, 0u, "Division by zero not allowed.");
+  const double helper = static_cast<double>(l2.y - l1.y) / static_cast<double>(l2.x - l1.x);
+  const typename TimestampCorrector<T>::time_t local_time =
+      static_cast<typename TimestampCorrector<T>::time_t>(
```

Comment:
The `static_cast<typename TimestampCorrector<T>::` (explicit scoping for time_t) should not be necessary here. This is only necessary for the return type of the function because that part of the function header is not yet within the scope of the class :).